doc/api/worker.md

```diff
@@ -0,0 +1,343 @@
+yields an object with `port1` and `port2` properties, which refer to linked
+[`MessagePort`][] instances.
+
+```js
+const { MessageChannel } = require('worker');
+
+const { port1, port2 } = new MessageChannel();
+port1.on('message', (message) => console.log('received', message));
+port2.postMessage({ foo: 'bar' });
+// prints: received { foo: 'bar' }
+```
+
+## Class: MessagePort
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+Instances of the `worker.MessagePort` class represent an asynchronous
+communications channel. It can be used to transfer structured data,
+memory regions and other `MessagePort`s between different [`Worker`][]s
+or [`vm` context][vm]s.
+
+For transferring `MessagePort` instances between VM contexts, see
+[`vm.moveMessagePortToContext()`][].
+
+*Note*: With the exception of `MessagePort`s being [`EventEmitter`][]s rather
+than `EventTarget`s, this implementation matches [browser `MessagePort`][]s.
+
+### Event: 'close'
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'close'` event is emitted once either side of the channel has been
+disconnected.
+
+### Event: 'error'
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'error'` event is emitted if the worker thread throws an uncaught
+expection. In that case, the worker will be terminated.
+
+### Event: 'message'
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'message'` event is emitted for any incoming message, containing the cloned
+input of [`port.postMessage()`][].
+
+### port.close()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {undefined}
+
+Disables further sending of messages on either side of the connection.
+
+### port.postMessage(value[, transferList])
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {undefined}
+
+Sends a JavaScript value to the receiving side of this channel.
+`value` will be transferred in a way
+that is compatible with the [HTML structured clone algorithm][]. In particular,
+it may contain circular references and objects like typed arrays that `JSON`
+is not able to serialize.
+
+`transferList` may be a list of `ArrayBuffer` and `MessagePort` objects.
+After transferring, they will not be usable on the sending side of the channel
+anymore.
+
+If `value` contains [`SharedArrayBuffer`][] instances, those will be accessible
+from either thread.
+
+*Note*: Because the object cloning uses the structured clone algorithm,
+non-enumberable properties, accessors, and prototypes are not preserved.
+In particular, [`Buffer`][] objects will be read as plain [`Uint8Array`][]s
+on the receiving side.
+
+### port.start()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {undefined}
+
+Starts receiving messages on this `MessagePort`. When using this port
+as an event emitter, this will be called automatically once `message` listeners
+are attached.
+
+
+## Class: Worker
+<!-- YAML
+added: REPLACEME
+-->
+
+The `Worker` class represents an independent JavaScript execution thread.
+Most Node APIs are available inside of it.
```

Comment:
Ayo or Ayo.js.

Is `Worker` available in a worker?

Reply:
> Is `Worker` available in a worker?

Yes. That’s also tested and the code is generally laid out to support such a structure, in part because there’s not really anything standing in the way of it